### PR TITLE
Allow customization of task per worker and add more printed info.

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

File: src/root_readspeed.cxx [hunk 1]
```diff
@@ -54,12 +54,18 @@ Args ParseArgs(int argc, char **argv)
          argState = EArgState::kBranches;
       else if (std::strcmp(argv[i], "--threads") == 0)
          argState = EArgState::kThreads;
-      else {
+      else if (std::strcmp(argv[i], "--tasksPerWorker") == 0)
```
Comment (eguiraud) [inline src/root_readspeed.cxx:57]:
```suggestion
      else if (std::strcmp(argv[i], "--tasks-per-worker") == 0)
```

**Review by eguiraud — COMMENTED** [review]
Thanks, LGTM. Before merging, @dkpoult can you please try out this patch and report whether it works as expected?

File: src/ReadSpeed.hxx [hunk 2]
```diff
@@ -203,6 +203,12 @@ inline Result EvalThroughputMT(const Data &d, unsigned nThreads)
    const auto rangesPerFile = MergeClusters(GetClusters(d), maxTasksPerFile);
    clsw.Stop();
 
+   size_t nranges = 0;
+   for(auto &r : rangesPerFile) {
+       nranges += r.size();
+   }
+   std::cout << "Total number of cluster ranges: " << nranges << "\n";
```
Comment (eguiraud) [inline src/ReadSpeed.hxx:210]: we are actually merging entry clusters together:

```suggestion
   std::cout << "Total number of entry ranges: " << nranges << "\n";
```

File: src/root_readspeed.cxx [hunk 3]
```diff
@@ -21,7 +21,7 @@ void PrintThroughput(const Result &r)
    std::cout << "CPU time:\t\t\t" << r.fCpuTime << " s\n";
    std::cout << "Uncompressed data read:\t\t" << r.fUncompressedBytesRead << " bytes\n";
 
-   std::cout << "Throughput:\t\t\t" << r.fUncompressedBytesRead / r.fRealTime / 1024 / 1024 << " MB/s\n";
+   std::cout << "Throughput:\t\t\t" << r.fUncompressedBytesRead / r.fRealTime / 1024 / 1024 << " MB/s\t" << r.fUncompressedBytesRead / r.fRealTime / 1024 / 1024 / r.fThreadPoolSize<< " MB/s/thread for " << r.fThreadPoolSize<< " threads\n";
```
Comment (dkpoult) [inline src/root_readspeed.cxx:24]:
```suggestion
   const auto effectiveThreads = std::max(r.fThreadPoolSize, 1u);
   std::cout << "Throughput:\t\t\t" << r.fUncompressedBytesRead / r.fRealTime / 1024 / 1024 << " MB/s\t" << r.fUncompressedBytesRead / r.fRealTime / 1024 / 1024 / effectiveThreads << " MB/s/thread for " << effectiveThreads << " threads\n";
```
Accounts for running in single threaded mode.

**Comment by eguiraud** [comment]
Superseded by #17

**Pull request closed without merging**